code/src/entrance.c

```diff
@@ -346,11 +346,32 @@ u8 EntranceCutscene_ShouldPlay(u8 flag) {
 
 void Entrance_CheckEpona() {
     s32 entrance = gGlobalContext->nextEntranceIndex;
-    s8 dest = gEntranceTable[entrance].scene;
-    //If Link is riding Epona but he's about to enter a scene where she can't spawn,
+    //If Link is riding Epona but he's about to go through an entrance where she can't spawn,
     //unset the Epona flag to avoid Master glitch, and restore temp B.
-    if (gSettingsContext.shuffleOverworldEntrances && entrance != 0x496 && (PLAYER->stateFlags1 & 0x00800000) &&
-        dest != 81 && dest != 87 && dest != 90 && dest != 93 && dest != 99) {
+    if (gSettingsContext.shuffleOverworldEntrances && (PLAYER->stateFlags1 & 0x00800000)) {
+
+        static const s16 validEponaEntrances[13] = {
```

Comment:
```suggestion
        static const s16 validEponaEntrances[] = {
```
Doing this and using `sizeof(validEponaEntrances) / sizeof(validEponaEntrances[0])` at the for loop will handle the size count automatically.

Reply:
Right, thanks